Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs

```diff
@@ -451,5 +448,43 @@ public KeyVaultConfigurationException(string message, Exception innerException)
             {
             }
         }
+
+        private static RetryPolicy CreateKeyVaultRetryPolicy()
+        {
+            return new RetryPolicy(
+                new KeyVaultRetryPolicy(),
+                new ExponentialBackoff(
+                    name: "KeyVaultExponentialBackoff",
+                    retryCount: 5,
+                    minBackoff: TimeSpan.FromSeconds(10),
+                    maxBackoff: TimeSpan.FromSeconds(60),
+                    deltaBackoff: TimeSpan.FromSeconds(10),
+                    firstFastRetry: false)); // All retries are subjects to the policy, even the first one
+        }
+
+        private sealed class KeyVaultRetryPolicy : ITransientErrorDetectionStrategy
+        {
+            /// <inheritdoc />
+            public bool IsTransient(Exception ex)
+            {
+                var message = ex.Message;
+
+                if (message.Contains("The remote name could not be resolved"))
+                {
+                    // In some cases, KeyVault provider may fail with HttpRequestException with an inner exception like 'The remote name could not be resolved: 'login.windows.net'.
+                    // Theoretically, this should be handled by the host, but to make error handling simple and consistent (this method throws one exception type) the handling is happening here.
+                    // This is a recoverable error.
+                    return true;
+                }
+
+                if (message.Contains("429"))

```

Comment:
>message.Contains("429") [](start = 20, length = 23)

This seems incredibly generic. Do we know the expected exception type? Or anything else within the body of the error message?

Reply:
It is KeyVaultErrorException but the exception type is internal.

The full message is: Operation returned an invalid status code '429'. And I can look for it.